doc/tutorial/index.rst

```diff
@@ -0,0 +1,213 @@
+.. _tutorial:
+
+===============
+Sphinx tutorial
+===============
+
+In this tutorial you will build a simple documentation project using Sphinx,
+and view it in your browser as HTML.
+The project will include narrative, handwritten documentation,
+as well as autogenerated API documentation.
+
+The tutorial is aimed towards Sphinx newcomers
+willing to learn the fundamentals of how projects are created and structured.
+You will create a fictional software library to generate random food recipes
+that will serve as a guide throughout the process,
+with the objective of properly documenting it.
+
+To showcase Sphinx capabilities for code documentation
+you will use Python,
+which also supports *automatic* documentation generation.
+
+.. note::
+
+   Several other languages are natively supported in Sphinx
+   for *manual* code documentation,
+   however they require extensions for *automatic* code documentation,
+   like `Breathe <https://breathe.readthedocs.io/>`_.
+
+To follow the instructions you will need access to a Linux-like command line
+and a basic understanding of how it works,
+as well as a working Python installation for development,
+since you will use *Python virtual environments* to create the project.
+
+Getting started
+---------------
+
+Setting up our project and development environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+On a new directory,
+create a file called ``README.rst``
+with the following contents:
+
+.. code-block:: rest
+
+   Lumache
+   =======
+
+   **Lumache** (/luˈmake/) is a Python library for cooks and food lovers
+   that creates recipes mixing random ingredients.
+
+It is a good moment to create a Python virtual environment
+and install the required tools.
+For that, open a command line terminal,
+``cd`` into the directory you just created,
+and run the following commands:
+
+.. code-block:: bash
+
+   $ python -m venv .venv
+   $ source .venv/bin/activate
+   (.venv) $ python -m pip install sphinx
+
+.. note::
+
+   The installation method used above
+   is described in more detail in :ref:`install-pypi`.
+   For the rest of this tutorial,
+   the instructions will assume a Python virtual environment.
+
+If you executed these instructions correctly,
+you should have the Sphinx command line tools available.
+You can do a basic verification running this command:
+
+.. code-block:: bash
+
+   (.venv) $ sphinx-build --version
+   sphinx-build 4.0.2
+
+If you see a similar output, you are on the right path!
+
+Creating the documentation layout
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Then, from the command line,
+run the following command:
+
+.. code-block:: bash
+
+   (.venv) $ sphinx-quickstart docs
+
+This will present you a series of questions
+required to create the basic directory and configuration layout for your project
+inside the `docs/` folder.
+To proceed, introduce these answers:
```

Comment:
I observed that you tend to remove colons at the end of sentences that end in "as follows", whereas you are okay with some other sentences that precede enumerations or instructions. I'd be keen to learn more, would you please share some explanations about it?

Reply:
English.SE seems to disagree https://english.stackexchange.com/a/394875/20057

Reply:
First I should preface that the dominant practice should be continued, but I don't know what that is for Sphinx. A style guide should be adopted. @tk0miya is there an established preferred style in Sphinx for how to end a sentence when followed by a list or code block?

There are varying opinions about ending a sentence with a colon or a period when followed by a _list_, but not many about following it with _code_. The context and audience need to be considered. For example, technical documentation versus a narrative tutorial. Even American and English English don't agree. No one way is correct, as they are all opinions. I may have been inconsistent when spotting these instances and reviewing them. Apologies for that.

Some guides prefer to end all sentences with a period and never a colon. I prefer this style because it prevents the confusion of what punctuation to use when ending a sentence, as they _always_ end with a period without exception. It avoids sentence fragments, such as "For example:" and encourages the writer to form complete sentences. This is especially helpful for visually impaired readers and screen readers. It prevents the confusion of whether to use `:` or `::` when a code block follows to get syntax highlighting.

"As follows", "following", and similar phrases are hints to readers—especially visually impaired or people with reading comprehension impairments—that something is coming up related to this sentence, so I almost always include them.

Reply:
>is there an established preferred style in Sphinx for how to end a sentence when followed by a list or code block?

AFAIK, there are no policies and rules for them.

IMO, reST provides `::` syntax to notate a code-block after a paragraph
```
blah blah blah::

    here is a code block.
```

This is rendered as following.
```
blah blah blah:

    here is a code block.
```

So I think the sentences that end with a colon just before code-block are commonly used in reST documents. I've usually written them also. To imitate the colon at the tail of the sentence, I've usually use a single colon and `code-block` directive as follows

```
blah blah blah:

.. code-block:: some-highlighting language

   here is a code block.
```

On the other hand, I don't see the sentences that end with a colon just before bullet-list much.